x/evm/keeper/statedb.go

```diff
@@ -522,6 +602,10 @@ func (k *Keeper) Exist(addr common.Address) bool {
 //
 // Non-ethereum accounts are considered not empty
 func (k *Keeper) Empty(addr common.Address) bool {
+	if k.HasStateError() {
+		return true
```

Comment:
why true? it should return the zero value (false) if there's an error

Reply:
Good Catch! Just updated.